bin/yarn.js

```diff
@@ -4,41 +4,24 @@
 /* eslint-disable flowtype/require-valid-file-annotation */
 'use strict';
 
-// validate that used node version is supported
-var semver = require('semver');
 var ver = process.versions.node;
-ver = ver.split('-')[0]; // explode and truncate tag from version #511
+var majorVer = parseInt(ver.split('.')[0], 10);
 
 var dirPath = null;
 
-if (semver.satisfies(ver, '>=5.0.0')) {
-  dirPath = '../lib';
-} else if (semver.satisfies(ver, '>=4.0.0')) {
-  dirPath = '../lib-legacy';
+if (majorVer >= 5) {
```

Comment:
We should not have any dependencies in this file anymore, hence all the changes. Also these new ones are simpler :)